napari/layers/base/base.py

```diff
@@ -1136,39 +1159,52 @@ def _slice_dims(
             it already has a valid slice ready. False otherwise.
         """
         logger.debug(
-            'Layer._slice_dims: %s, point=%s, ndisplay=%s, order=%s, force=%s',
+            'Layer._slice_dims: %s, point=%s, margin_left=%s, margin_right=%s, ndisplay=%s, order=%s, force=%s',
             self,
             point,
+            margin_left,
+            margin_right,
             ndisplay,
             order,
             force,
         )
-        slice_input = self._make_slice_input(point, ndisplay, order)
+        slice_input = self._make_slice_input(
+            point, margin_left, margin_right, ndisplay, order
+        )
         if force or (self._slice_input != slice_input):
             self._slice_input = slice_input
             self._refresh_sync()
 
     def _make_slice_input(
-        self, point=None, ndisplay=2, order=None
+        self,
+        point=None,
+        margin_left=None,
+        margin_right=None,
+        ndisplay=2,
+        order=None,
     ) -> _SliceInput:
-        point = (0,) * self.ndim if point is None else tuple(point)
-
-        ndim = len(point)
+        # if not point is given, "world" has same dimensionality of self
+        world_ndim = self.ndim if point is None else len(point)
 
-        if order is None:
-            order = tuple(range(ndim))
+        world_slice = _ThickNDSlice.make_full(
+            point=point,
+            margin_left=margin_left,
+            margin_right=margin_right,
+            ndim=world_ndim,
+        )
 
-        # Correspondence between dimensions across all layers and
-        # dimensions of this layer.
-        point = point[-self.ndim :]
+        order = tuple(range(world_ndim)) if order is None else tuple(order)
         order = tuple(
-            self._world_to_layer_dims(world_dims=order, ndim_world=ndim)
+            self._world_to_layer_dims(
+                world_dims=order,
+                ndim_world=world_ndim,
+            )
         )
 
         return _SliceInput(
             ndisplay=ndisplay,
-            point=point,
-            order=order,
+            world_slice=world_slice[-self.ndim :],
```

Comment:
Does `_ThickNDSlice.make_full` handle the world-to-layer dimensional correspondence that `_world_to_layer_dims` handles? Does it do in a different way? We already have a few functions that do this slightly differently (maybe for good reasons), but it would be nice to avoid different definitions and code/logic reuse if not needed.

Reply:
`make_full` doesn't do any coordinate space transformation. It's staying in whatever coordinate space the inputs are. In fact, it's used both as a world space slice in `_SliceInput` and as a data space slice in the `SliceRequest` classes.

Reply:
Oh I see, `order` is still used to handle that transformation. Thanks for clarifying!